x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/data_stream_step/data_stream_step.tsx

```diff
@@ -0,0 +1,231 @@
+  ({ integrationSettings, connectorId, isGenerating }) => {
+    const { setIntegrationSettings, setIsGenerating, setStep, setResult } = useActions();
+    const { isLoading: isLoadingPackageNames, packageNames } = useLoadPackageNames();
+
+    const [name, setName] = useState<string>(integrationSettings?.name ?? '');
+    const [dataStreamName, setDataStreamName] = useState<string>(
+      integrationSettings?.dataStreamName ?? ''
+    );
+    const [invalidFields, setInvalidFields] = useState({ name: false, dataStreamName: false });
+
+    const setIntegrationValues = useCallback(
+      (settings: Partial<IntegrationSettings>) =>
+        setIntegrationSettings({ ...integrationSettings, ...settings }),
+      [integrationSettings, setIntegrationSettings]
+    );
+
+    const onChange = useMemo(() => {
+      return {
+        name: (e: React.ChangeEvent<HTMLInputElement>) => {
+          const nextName = e.target.value;
+          setName(nextName);
+          if (!isValidName(nextName) || packageNames?.has(nextName)) {
+            setInvalidFields((current) => ({ ...current, name: true }));
+            setIntegrationValues({ name: undefined });
+          } else {
+            setInvalidFields((current) => ({ ...current, name: false }));
+            setIntegrationValues({ name: nextName });
+          }
+        },
+        dataStreamName: (e: React.ChangeEvent<HTMLInputElement>) => {
+          const nextDataStreamName = e.target.value;
+          setDataStreamName(nextDataStreamName);
+          if (!isValidName(nextDataStreamName)) {
+            setInvalidFields((current) => ({ ...current, dataStreamName: true }));
+            setIntegrationValues({ dataStreamName: undefined });
+          } else {
+            setInvalidFields((current) => ({ ...current, dataStreamName: false }));
+            setIntegrationValues({ dataStreamName: nextDataStreamName });
+          }
+        },
+        // inputs without validation
+        dataStreamTitle: (e: React.ChangeEvent<HTMLInputElement>) =>
+          setIntegrationValues({ dataStreamTitle: e.target.value }),
+        dataStreamDescription: (e: React.ChangeEvent<HTMLInputElement>) =>
+          setIntegrationValues({ dataStreamDescription: e.target.value }),
+        inputType: (e: React.ChangeEvent<HTMLSelectElement>) => {
+          setIntegrationValues({ inputType: e.target.value as InputType });
+        },
+      };
+    }, [setIntegrationValues, setInvalidFields, packageNames]);
+
+    // pre-populate the name from title
+    useEffect(() => {
+      if (packageNames != null) {
+        if (integrationSettings?.title && integrationSettings.name == null) {
+          const generatedName = getNameFromTitle(integrationSettings.title);
+          if (!packageNames.has(generatedName)) {
+            setName(generatedName);
+            setIntegrationValues({ name: generatedName });
+          }
+        }
+      }
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Good point, but It does not break, actually this is intentional.

I only want to try to pre-populate the package _name_ once, when `packageNames` is defined (loaded using fleet epm API), and whatever the outcome is, populated or not (duplicate name) I don't want to execute this logic anymore.

If I put all the dependencies it will try to pre-populate the name on every state update. The result will be the same every time, but that's unnecessary executions that can be avoided with controlled dependencies.

I wish I could avoid using _useEffect_, but that's the only way here.

Reply:
https://github.com/elastic/kibana/commit/f2fe6207796026eb8d3b935b3270be3a43f7c9c2#diff-fa374325098a7b72d48803bdaacc8ed2684655e97a80d2ac2ffa2014ee2e294cR83

Reply:
https://github.com/elastic/kibana/commit/f2fe6207796026eb8d3b935b3270be3a43f7c9c2#diff-a89884afb36cc67c03c957120ac4ba2c5141210665f75564d87d8ebdfcc2ef39R105-R106